syntax = "proto3";
package collector_job;





/*
CreateCollectorJob请求
*/
message CreateCollectorJobRequest {
message CollectTime {
    /*
    采集开始时间
    */
    string startTime = 1 ;
    /*
    采集结束时间
    */
    string endTime = 2 ;
}
        
    
    /*
    关联的采集模版ID
    */
    string collectorTemplateId = 1 ;
    /*
    模型id
    */
    string objectId = 2 ;
    /*
    任务名称
    */
    string jobName = 3 ;
    /*
    过滤条件
    */
    string filter = 4 ;
    /*
    采集时段
    */
    CollectTime collectTime = 5 ;
    /*
    执行超时
    */
    int32 timeout = 6 ;
    /*
    是否启用
    */
    bool enabled = 7 ;
    /*
    采集间隔
    */
    int32 interval = 8 ;
    /*
    关系查询路径
    */
    string path = 9 ;
    /*
    关系查询路径ID
    */
    string pathId = 10 ;
}

/*
CreateCollectorJob返回
*/
message CreateCollectorJobResponse {
    /*
    实例ID
    */
    string instanceId = 1 ;
}

/*
CreateCollectorJobApi返回
*/
message CreateCollectorJobResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    collector_job.CreateCollectorJobResponse data = 4 ;
}

